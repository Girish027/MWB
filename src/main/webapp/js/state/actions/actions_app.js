
import _ from 'lodash';
import getUrl, { pathKey } from 'utils/apiUrls';
import { fetchUtil } from 'utils/fetchUtils';
import goToRoute from 'utils/routeHelpers';
import Constants from 'constants/Constants';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { logAmplitudeEvent } from 'utils/amplitudeUtils';
import axios from 'axios';
import api from 'utils/api';
import { setDefaultsForAPI } from 'utils/api';
import * as types from './types';

// Note to developers:
// Both Warning and Good Messages are shown for the period of notificationInterval.
// Bad, Persistant - till user closes the notification. This should only be used sparingly.
export const callLogIngestAPI = (message, logLevel) => {
  const promise = new Promise((resolve, reject) => {
    const url = getUrl(pathKey.logIngest, { logLevel });
    return new Promise((resolve, reject) => {
      api.post({
        url,
        headers: {
          'Content-Type': 'text/plain',
        },
        data: message,
      });
    });
  });
};

export const displayBadRequestMessage = (message) => (dispatch) => {
  dispatch({
    type: types.BAD_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_ERROR);
};

export const displayWarningRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.WARNING_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_WARNING);

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayGoodRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.GOOD_REQUEST,
    value: message,
  });

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayPersistantMessage = message => ({
  type: types.PERSISTENT_REQUEST,
  value: message,
});

export const stopShowingServerMessage = () => ({
  type: types.APP_STOP_SHOWING_SERVER_MESSAGE,
});

export const updateLoginSuccess = csrfToken => ({
  type: types.LOGIN_SUCCESS,
  value: csrfToken,
});

export const updateLoginInfo = (userId, userName, userDetails) => ({
  type: types.LOGIN_INFO,
  value: {
    userId,
    userName,
    userDetails,
  },
});

export const logoutFromOktaSuccess = () => ({
  type: types.LOGOUT_OKTA,
});

export const logoutFromMWBSuccess = () => ({
  type: types.LOGOUT_MWB,
});

export const contextMenuChange = contextMenuState => ({
  type: types.CONTEXT_MENU_CHANGE,
  contextMenuState,
});

export const modalDialogChange = modalDialogState => ({
  type: types.MODAL_DIALOG_CHANGE,
  modalDialogState,
});

const actionFileRemoveRequest = ({ fileId }) => ({
  type: types.FILE_REMOVE_REQUEST,
  fileId,
});

const actionFileRemoveError = ({ fileId, error }) => ({
  type: types.FILE_REMOVE_ERROR,
  fileId,
  error,
});

const actionFileRemoveSuccess = ({ fileId }) => ({
  type: types.FILE_REMOVE_SUCCESS,
  fileId,
});

export const updateUserGroups = data => ({
  type: types.UPDATE_USER_GROUPS,
  data,
});

const onFetchFailureRemoveFile = (error, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveError({ fileId, error }));
};

const onFetchSuccessRemoveFile = (data, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveSuccess({ fileId }));
};

export const removeFile = ({ fileId }) => (dispatch, getState) => {
  dispatch(actionFileRemoveRequest({ fileId }));

  const state = getState();
  const fetchUrl = getUrl(pathKey.files, { fileId, clientId: state.header.client.id });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'delete',
    onFetchSuccess: onFetchSuccessRemoveFile,
    onFetchFailure: onFetchFailureRemoveFile,
    dispatch,
    getState,
    params: {
      fileId,
    },
  });
};

const onFetchSuccessGetUserGroups = data => (dispatch) => {
  dispatch(updateUserGroups(data));
};

export const getUserGroups = ({ csrfToken }) => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.userGroups, { csrfToken });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetUserGroups,
    dispatch,
    getState,
  });
};

export const onFetchSuccessGetCSRFToken = data => (dispatch, getState) => {
  const csrfToken = data.value;
  setDefaultsForAPI(csrfToken);
  dispatch(updateLoginSuccess(csrfToken));
  dispatch(getUserGroups({ csrfToken }));
};

export const getCSRFToken = userId => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.csrfToken);

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetCSRFToken,
    dispatch,
    getState,
    params: {
      userId,
    },
  });
};

const logoutFromMWB = () => (dispatch, getState) => {
  dispatch(logoutFromOktaSuccess());
  dispatch(logoutFromMWBSuccess());
};

export const logoutFromOkta = () => (dispatch, getState) => {
  const state = getState();
  const { oAuthLogoutURL } = state.app;
  if (oAuthLogoutURL) {
    fetch(oAuthLogoutURL, {
      method: 'DELETE',
      credentials: 'include',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    })
      .then((response) => {
        window.location.href = Constants.LOGOUT;
        switch (response.status) {
        case 204:
          logoutFromMWB();
          logAmplitudeEvent(AmplitudeConstants.LOGOUT_EVENT, state);
          break;
        case 404:
        default:
          // TODO send log for these scenarios
          break;
        }
      })
      .catch((error) => {
        // TODO send log for these scenarios
      });
  }
};

export const changeRoute = (routeName, data, history) => (dispatch, getState) => {
  const state = getState();
  const routeData = Object.assign({}, {
    client: !_.isNil(state.header.client) ? state.header.client : {
      id: 0,
      name: 'Loading...',
    },
    projectId: state.selectedProjectId,
  }, data);
  goToRoute(routeName, routeData, history);
};
