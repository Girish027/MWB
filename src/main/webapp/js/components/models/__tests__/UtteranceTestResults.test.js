import React from 'react';
import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import UtteranceTestResults from 'components/models/UtteranceTestResults';
import * as actionsModels from 'state/actions/actions_models';
import Constants from 'constants/Constants';

describe('<UtteranceTestResults />', () => {
  const digitalModelTestResults = {
    projectId: '661',
    modelId: 'b61693f6-fa65-4308-af09-b32872dfb5f6',
    type: 'UTTERANCES',
    status: 'SUCCESS',
    evaluations: [{
      utterance: 'i want to cancel for june 13',
      intents: [{
        intent: 'reservation-query',
        score: 0.49544276870123316,
      }, {
        intent: 'agent-query',
        score: 0.42004871402243316,
      }, {
        intent: 'None_None',
        score: 0.14285217997731586,
      }],
      transformations: [{
        id: 'rx-mapper.non-breaking-space-regex',
        result: 'i want to cancel for june 13',
      }, {
        id: 'wsp',
        result: 'i want to cancel for june 13',
      }],
      entities: [{
        name: 'date',
        value: 'june 13',
      }],
    }],
  };

  const speechModelTestResults = {
    projectId: '661',
    modelId: 'b61693f6-fa65-4308-af09-b32872dfb5f6',
    type: 'UTTERANCES',
    status: 'SUCCESS',
    evaluations: [{
      recognitionScore: '0.9312356',
      utterance: 'i want to cancel for june 13',
      utteranceWithWordClass: 'i want to cancel for speech_class_date',
      intents: [{
        intent: 'speech-intent',
        score: 0.49544276870123316,
      }, {
        intent: 'speech-intent2',
        score: 0.42004871402243316,
      }],
    }],
  };

  const propsDigital = {
    digitalModelTestResults,
    testModelType: Constants.DIGITAL_MODEL,
  };

  const propsSpeech = {
    speechModelTestResults,
    digitalModelTestResults,
    testModelType: Constants.DIGITAL_SPEECH_MODEL,
    fileType: Constants.FILE_TYPE.LINK,
  };

  let wrapper;

  afterAll(() => {
    jest.clearAllMocks();
  });

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = shallow(<UtteranceTestResults
        {...propsDigital}
      />);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly for a digital model when modelTestResults are present', () => {
      wrapper = shallow(<UtteranceTestResults
        {...propsDigital}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly for a speech model when modelTestResults are present', () => {
      wrapper = shallow(<UtteranceTestResults
        {...propsSpeech}
      />);
      wrapper.instance().getFileName = jest.fn(() => 'audio_1557955235157.wav');
      wrapper.setState({
        activeTab: Constants.TEST_RESULTS_TABS.entitiesTab,
      });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly for a digital model when modelTestResults are absent', () => {
      wrapper = shallow(<UtteranceTestResults
        {...propsDigital}
        digitalModelTestResults={{}}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly for a speech model when modelTestResults are absent', () => {
      wrapper = shallow(<UtteranceTestResults
        {...propsSpeech}
        digitalModelTestResults={{}}
        speechModelTestResults={{}}
      />);
      wrapper.instance().getFileName = jest.fn(() => 'audio_1557955235157.wav');
      wrapper.setState({
        activeTab: Constants.TEST_RESULTS_TABS.entitiesTab,
      });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly modelTestResults are does not have entities', () => {
      wrapper = shallow(<UtteranceTestResults
        {...propsSpeech}
        digitalModelTestResults={{
          ...digitalModelTestResults,
          evaluations: [{
            utterance: 'i want to cancel for june 13',
            intents: [{
              intent: 'reservation-query',
              score: 0.49544276870123316,
            }],
            transformations: [{
              id: 'rx-mapper.non-breaking-space-regex',
              result: 'i want to cancel for june 13',
            }],
            entities: [],
          }],
        }
        }
      />);
      wrapper.instance().getFileName = jest.fn(() => 'audio_1557955235157.wav');
      wrapper.setState({
        activeTab: Constants.TEST_RESULTS_TABS.entitiesTab,
      });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly modelTestResults are does not have transformations', () => {
      wrapper = shallow(<UtteranceTestResults
        {...propsSpeech}
        digitalModelTestResults={{
          ...digitalModelTestResults,
          evaluations: [{
            utterance: 'i want to cancel for june 13',
            intents: [{
              intent: 'reservation-query',
              score: 0.49544276870123316,
            }],
            transformations: [],
            entities: [{
              name: 'date',
              value: 'june 13',
            }],
          }],
        }
        }
      />);
      wrapper.instance().getFileName = jest.fn(() => 'audio_1557955235157.wav');
      wrapper.setState({
        activeTab: Constants.TEST_RESULTS_TABS.entitiesTab,
      });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly modelTestResults are does not have intents', () => {
      wrapper = shallow(<UtteranceTestResults
        {...propsSpeech}
        digitalModelTestResults={{
          ...digitalModelTestResults,
          evaluations: [{
            utterance: 'i want to cancel for june 13',
            intents: [],
            transformations: [{
              id: 'rx-mapper.non-breaking-space-regex',
              result: 'i want to cancel for june 13',
            }],
            entities: [{
              name: 'date',
              value: 'june 13',
            }],
          }],
        }
        }
      />);
      wrapper.instance().getFileName = jest.fn(() => 'audio_1557955235157.wav');
      wrapper.setState({
        activeTab: Constants.TEST_RESULTS_TABS.entitiesTab,
      });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly when a tab currently active', () => {
      wrapper = shallow(<UtteranceTestResults
        {...propsSpeech}
      />);
      wrapper.instance().getFileName = jest.fn(() => 'audio_1557955235157.wav');
      wrapper.setState({
        activeTab: Constants.TEST_RESULTS_TABS.entitiesTab,
      });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    afterEach(() => {
      jest.clearAllMocks();
    });

    describe('getFileName:', () => {
      test('should get the name of the file with current timestamp', () => {
        expect(wrapper.instance().getFileName().match(/audio_\d+\.wav/)).not.toEqual(null);
      });
    });

    describe('onTabSelected:', () => {
      test('should set the state to match the new selected index', () => {
        wrapper.setState({
          selectedTabIndex: 1,
        });
        wrapper.instance().onTabSelected('Trasnformations Tab', 0);
        expect(wrapper.state().selectedTabIndex).toEqual(0);
      });
    });
  });
});
