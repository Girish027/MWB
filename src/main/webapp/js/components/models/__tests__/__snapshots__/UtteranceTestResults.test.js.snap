// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UtteranceTestResults /> Snapshots renders correctly for a digital model when modelTestResults are absent 1`] = `
<Deprecated
  styleOverride={
    Object {
      "borderTop": "1px solid #dadadd",
    }
  }
>
  <Placeholder
    message="Type an utterance to run a test"
    styleOverride={Object {}}
  />
</Deprecated>
`;

exports[`<UtteranceTestResults /> Snapshots renders correctly for a digital model when modelTestResults are present 1`] = `
<Deprecated
  styleOverride={
    Object {
      "borderTop": "1px solid #dadadd",
    }
  }
>
  <span>
    <withTheme(Tabs)
      align="left"
      onTabSelected={[Function]}
      selectedIndex={0}
      tabPanels={
        Array [
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestIntentsGrid
              data={
                Array [
                  Object {
                    "intent": "reservation-query",
                    "score": 0.49544276870123316,
                  },
                  Object {
                    "intent": "agent-query",
                    "score": 0.42004871402243316,
                  },
                  Object {
                    "intent": "None_None",
                    "score": 0.14285217997731586,
                  },
                ]
              }
            />
          </Deprecated>,
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestTransformsGrid
              data={
                Array [
                  Object {
                    "id": "rx-mapper.non-breaking-space-regex",
                    "result": "i want to cancel for june 13",
                  },
                  Object {
                    "id": "wsp",
                    "result": "i want to cancel for june 13",
                  },
                ]
              }
            />
          </Deprecated>,
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestEntitiesGrid
              data={
                Array [
                  Object {
                    "name": "date",
                    "value": "june 13",
                  },
                ]
              }
            />
          </Deprecated>,
        ]
      }
      tabs={
        Array [
          "Predicted Intents (3)",
          "Transformations (2)",
          "Entities (1)",
        ]
      }
    />
  </span>
</Deprecated>
`;

exports[`<UtteranceTestResults /> Snapshots renders correctly for a speech model when modelTestResults are absent 1`] = `
<Deprecated
  styleOverride={
    Object {
      "borderTop": "1px solid #dadadd",
    }
  }
>
  <Placeholder
    message="Select, record, upload, or link an audio file to run a test"
    styleOverride={Object {}}
  />
</Deprecated>
`;

exports[`<UtteranceTestResults /> Snapshots renders correctly for a speech model when modelTestResults are present 1`] = `
<Deprecated
  styleOverride={
    Object {
      "borderTop": "1px solid #dadadd",
    }
  }
>
  <span>
    <Deprecated
      styleOverride={
        Object {
          "borderBottom": "1px solid #dadadd",
        }
      }
    >
      <ul
        className="utterance-reco-score-ul"
      >
        <li
          className="center-item-li recognition-score"
          key="recognition-score"
        >
          Recognition Score
          <span
            className="recognition-score-data"
          >
             
            93.12 %
             
          </span>
        </li>
        <li
          className="center-item-li word-class"
        >
          <div>
            <span
              className="utterance-result"
            >
              i want to cancel for june 13
            </span>
            <AudioPlayer
              audioSrc=""
              audioStates={Object {}}
              dispatch={[Function]}
              iconProps={
                Object {
                  "common": Object {
                    "circleStroke": "#004C97",
                    "height": "20px",
                    "pathFill": "#004C97",
                    "width": "20px",
                  },
                }
              }
              onPlay={[Function]}
              onStop={[Function]}
              playIcon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                }
              }
              stopIcon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                }
              }
              styleOverride={Object {}}
            />
            <span
              style={
                Object {
                  "paddingLeft": "0px",
                  "paddingRight": "20px",
                }
              }
            >
              <Downloader
                file=""
                fileName="audio_1557955235157.wav"
                fileType="link"
                iconProps={Object {}}
              >
                <Memo(withTheme(Download))
                  fill="#004C97"
                  height="16px"
                  width="12px"
                />
              </Downloader>
            </span>
          </div>
          <span>
             Utterance with word class:
          </span>
          <span
            style={
              Object {
                "fontWeight": "normal",
                "paddingLeft": "5px",
              }
            }
          >
            i want to cancel for speech_class_date
          </span>
        </li>
      </ul>
    </Deprecated>
    <withTheme(Tabs)
      align="left"
      onTabSelected={[Function]}
      selectedIndex={0}
      tabPanels={
        Array [
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestIntentsGrid
              data={
                Array [
                  Object {
                    "intent": "speech-intent",
                    "score": 0.49544276870123316,
                  },
                  Object {
                    "intent": "speech-intent2",
                    "score": 0.42004871402243316,
                  },
                ]
              }
            />
          </Deprecated>,
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestTransformsGrid
              data={
                Array [
                  Object {
                    "id": "rx-mapper.non-breaking-space-regex",
                    "result": "i want to cancel for june 13",
                  },
                  Object {
                    "id": "wsp",
                    "result": "i want to cancel for june 13",
                  },
                ]
              }
            />
          </Deprecated>,
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestEntitiesGrid
              data={
                Array [
                  Object {
                    "name": "date",
                    "value": "june 13",
                  },
                ]
              }
            />
          </Deprecated>,
        ]
      }
      tabs={
        Array [
          "Predicted Intents (2)",
          "Transformations (2)",
          "Entities (1)",
        ]
      }
    />
  </span>
</Deprecated>
`;

exports[`<UtteranceTestResults /> Snapshots renders correctly modelTestResults are does not have entities 1`] = `
<Deprecated
  styleOverride={
    Object {
      "borderTop": "1px solid #dadadd",
    }
  }
>
  <span>
    <Deprecated
      styleOverride={
        Object {
          "borderBottom": "1px solid #dadadd",
        }
      }
    >
      <ul
        className="utterance-reco-score-ul"
      >
        <li
          className="center-item-li recognition-score"
          key="recognition-score"
        >
          Recognition Score
          <span
            className="recognition-score-data"
          >
             
            93.12 %
             
          </span>
        </li>
        <li
          className="center-item-li word-class"
        >
          <div>
            <span
              className="utterance-result"
            >
              i want to cancel for june 13
            </span>
            <AudioPlayer
              audioSrc=""
              audioStates={Object {}}
              dispatch={[Function]}
              iconProps={
                Object {
                  "common": Object {
                    "circleStroke": "#004C97",
                    "height": "20px",
                    "pathFill": "#004C97",
                    "width": "20px",
                  },
                }
              }
              onPlay={[Function]}
              onStop={[Function]}
              playIcon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                }
              }
              stopIcon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                }
              }
              styleOverride={Object {}}
            />
            <span
              style={
                Object {
                  "paddingLeft": "0px",
                  "paddingRight": "20px",
                }
              }
            >
              <Downloader
                file=""
                fileName="audio_1557955235157.wav"
                fileType="link"
                iconProps={Object {}}
              >
                <Memo(withTheme(Download))
                  fill="#004C97"
                  height="16px"
                  width="12px"
                />
              </Downloader>
            </span>
          </div>
          <span>
             Utterance with word class:
          </span>
          <span
            style={
              Object {
                "fontWeight": "normal",
                "paddingLeft": "5px",
              }
            }
          >
            i want to cancel for speech_class_date
          </span>
        </li>
      </ul>
    </Deprecated>
    <withTheme(Tabs)
      align="left"
      onTabSelected={[Function]}
      selectedIndex={0}
      tabPanels={
        Array [
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestIntentsGrid
              data={
                Array [
                  Object {
                    "intent": "speech-intent",
                    "score": 0.49544276870123316,
                  },
                  Object {
                    "intent": "speech-intent2",
                    "score": 0.42004871402243316,
                  },
                ]
              }
            />
          </Deprecated>,
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestTransformsGrid
              data={
                Array [
                  Object {
                    "id": "rx-mapper.non-breaking-space-regex",
                    "result": "i want to cancel for june 13",
                  },
                ]
              }
            />
          </Deprecated>,
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <Placeholder
              message="No Entities Results"
              styleOverride={Object {}}
            />
          </Deprecated>,
        ]
      }
      tabs={
        Array [
          "Predicted Intents (2)",
          "Transformations (1)",
          "Entities (0)",
        ]
      }
    />
  </span>
</Deprecated>
`;

exports[`<UtteranceTestResults /> Snapshots renders correctly modelTestResults are does not have intents 1`] = `
<Deprecated
  styleOverride={
    Object {
      "borderTop": "1px solid #dadadd",
    }
  }
>
  <span>
    <Deprecated
      styleOverride={
        Object {
          "borderBottom": "1px solid #dadadd",
        }
      }
    >
      <ul
        className="utterance-reco-score-ul"
      >
        <li
          className="center-item-li recognition-score"
          key="recognition-score"
        >
          Recognition Score
          <span
            className="recognition-score-data"
          >
             
            93.12 %
             
          </span>
        </li>
        <li
          className="center-item-li word-class"
        >
          <div>
            <span
              className="utterance-result"
            >
              i want to cancel for june 13
            </span>
            <AudioPlayer
              audioSrc=""
              audioStates={Object {}}
              dispatch={[Function]}
              iconProps={
                Object {
                  "common": Object {
                    "circleStroke": "#004C97",
                    "height": "20px",
                    "pathFill": "#004C97",
                    "width": "20px",
                  },
                }
              }
              onPlay={[Function]}
              onStop={[Function]}
              playIcon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                }
              }
              stopIcon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                }
              }
              styleOverride={Object {}}
            />
            <span
              style={
                Object {
                  "paddingLeft": "0px",
                  "paddingRight": "20px",
                }
              }
            >
              <Downloader
                file=""
                fileName="audio_1557955235157.wav"
                fileType="link"
                iconProps={Object {}}
              >
                <Memo(withTheme(Download))
                  fill="#004C97"
                  height="16px"
                  width="12px"
                />
              </Downloader>
            </span>
          </div>
          <span>
             Utterance with word class:
          </span>
          <span
            style={
              Object {
                "fontWeight": "normal",
                "paddingLeft": "5px",
              }
            }
          >
            i want to cancel for speech_class_date
          </span>
        </li>
      </ul>
    </Deprecated>
    <withTheme(Tabs)
      align="left"
      onTabSelected={[Function]}
      selectedIndex={0}
      tabPanels={
        Array [
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestIntentsGrid
              data={
                Array [
                  Object {
                    "intent": "speech-intent",
                    "score": 0.49544276870123316,
                  },
                  Object {
                    "intent": "speech-intent2",
                    "score": 0.42004871402243316,
                  },
                ]
              }
            />
          </Deprecated>,
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestTransformsGrid
              data={
                Array [
                  Object {
                    "id": "rx-mapper.non-breaking-space-regex",
                    "result": "i want to cancel for june 13",
                  },
                ]
              }
            />
          </Deprecated>,
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestEntitiesGrid
              data={
                Array [
                  Object {
                    "name": "date",
                    "value": "june 13",
                  },
                ]
              }
            />
          </Deprecated>,
        ]
      }
      tabs={
        Array [
          "Predicted Intents (2)",
          "Transformations (1)",
          "Entities (1)",
        ]
      }
    />
  </span>
</Deprecated>
`;

exports[`<UtteranceTestResults /> Snapshots renders correctly modelTestResults are does not have transformations 1`] = `
<Deprecated
  styleOverride={
    Object {
      "borderTop": "1px solid #dadadd",
    }
  }
>
  <span>
    <Deprecated
      styleOverride={
        Object {
          "borderBottom": "1px solid #dadadd",
        }
      }
    >
      <ul
        className="utterance-reco-score-ul"
      >
        <li
          className="center-item-li recognition-score"
          key="recognition-score"
        >
          Recognition Score
          <span
            className="recognition-score-data"
          >
             
            93.12 %
             
          </span>
        </li>
        <li
          className="center-item-li word-class"
        >
          <div>
            <span
              className="utterance-result"
            >
              i want to cancel for june 13
            </span>
            <AudioPlayer
              audioSrc=""
              audioStates={Object {}}
              dispatch={[Function]}
              iconProps={
                Object {
                  "common": Object {
                    "circleStroke": "#004C97",
                    "height": "20px",
                    "pathFill": "#004C97",
                    "width": "20px",
                  },
                }
              }
              onPlay={[Function]}
              onStop={[Function]}
              playIcon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                }
              }
              stopIcon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                }
              }
              styleOverride={Object {}}
            />
            <span
              style={
                Object {
                  "paddingLeft": "0px",
                  "paddingRight": "20px",
                }
              }
            >
              <Downloader
                file=""
                fileName="audio_1557955235157.wav"
                fileType="link"
                iconProps={Object {}}
              >
                <Memo(withTheme(Download))
                  fill="#004C97"
                  height="16px"
                  width="12px"
                />
              </Downloader>
            </span>
          </div>
          <span>
             Utterance with word class:
          </span>
          <span
            style={
              Object {
                "fontWeight": "normal",
                "paddingLeft": "5px",
              }
            }
          >
            i want to cancel for speech_class_date
          </span>
        </li>
      </ul>
    </Deprecated>
    <withTheme(Tabs)
      align="left"
      onTabSelected={[Function]}
      selectedIndex={0}
      tabPanels={
        Array [
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestIntentsGrid
              data={
                Array [
                  Object {
                    "intent": "speech-intent",
                    "score": 0.49544276870123316,
                  },
                  Object {
                    "intent": "speech-intent2",
                    "score": 0.42004871402243316,
                  },
                ]
              }
            />
          </Deprecated>,
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <Placeholder
              message="No Transformations Results"
              styleOverride={Object {}}
            />
          </Deprecated>,
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestEntitiesGrid
              data={
                Array [
                  Object {
                    "name": "date",
                    "value": "june 13",
                  },
                ]
              }
            />
          </Deprecated>,
        ]
      }
      tabs={
        Array [
          "Predicted Intents (2)",
          "Transformations (0)",
          "Entities (1)",
        ]
      }
    />
  </span>
</Deprecated>
`;

exports[`<UtteranceTestResults /> Snapshots renders correctly when a tab currently active 1`] = `
<Deprecated
  styleOverride={
    Object {
      "borderTop": "1px solid #dadadd",
    }
  }
>
  <span>
    <Deprecated
      styleOverride={
        Object {
          "borderBottom": "1px solid #dadadd",
        }
      }
    >
      <ul
        className="utterance-reco-score-ul"
      >
        <li
          className="center-item-li recognition-score"
          key="recognition-score"
        >
          Recognition Score
          <span
            className="recognition-score-data"
          >
             
            93.12 %
             
          </span>
        </li>
        <li
          className="center-item-li word-class"
        >
          <div>
            <span
              className="utterance-result"
            >
              i want to cancel for june 13
            </span>
            <AudioPlayer
              audioSrc=""
              audioStates={Object {}}
              dispatch={[Function]}
              iconProps={
                Object {
                  "common": Object {
                    "circleStroke": "#004C97",
                    "height": "20px",
                    "pathFill": "#004C97",
                    "width": "20px",
                  },
                }
              }
              onPlay={[Function]}
              onStop={[Function]}
              playIcon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                }
              }
              stopIcon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                }
              }
              styleOverride={Object {}}
            />
            <span
              style={
                Object {
                  "paddingLeft": "0px",
                  "paddingRight": "20px",
                }
              }
            >
              <Downloader
                file=""
                fileName="audio_1557955235157.wav"
                fileType="link"
                iconProps={Object {}}
              >
                <Memo(withTheme(Download))
                  fill="#004C97"
                  height="16px"
                  width="12px"
                />
              </Downloader>
            </span>
          </div>
          <span>
             Utterance with word class:
          </span>
          <span
            style={
              Object {
                "fontWeight": "normal",
                "paddingLeft": "5px",
              }
            }
          >
            i want to cancel for speech_class_date
          </span>
        </li>
      </ul>
    </Deprecated>
    <withTheme(Tabs)
      align="left"
      onTabSelected={[Function]}
      selectedIndex={0}
      tabPanels={
        Array [
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestIntentsGrid
              data={
                Array [
                  Object {
                    "intent": "speech-intent",
                    "score": 0.49544276870123316,
                  },
                  Object {
                    "intent": "speech-intent2",
                    "score": 0.42004871402243316,
                  },
                ]
              }
            />
          </Deprecated>,
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestTransformsGrid
              data={
                Array [
                  Object {
                    "id": "rx-mapper.non-breaking-space-regex",
                    "result": "i want to cancel for june 13",
                  },
                  Object {
                    "id": "wsp",
                    "result": "i want to cancel for june 13",
                  },
                ]
              }
            />
          </Deprecated>,
          <Deprecated
            styleOverride={
              Object {
                "paddingBottom": "30px",
                "paddingTop": "30px",
              }
            }
          >
            <ModelTestEntitiesGrid
              data={
                Array [
                  Object {
                    "name": "date",
                    "value": "june 13",
                  },
                ]
              }
            />
          </Deprecated>,
        ]
      }
      tabs={
        Array [
          "Predicted Intents (2)",
          "Transformations (2)",
          "Entities (1)",
        ]
      }
    />
  </span>
</Deprecated>
`;
